src/compiler/checker.ts

```diff
@@ -18087,14 +18087,7 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
         if (contains(types, wildcardType)) {
             return wildcardType;
         }
-        if (
-            texts.length === 2 && texts[0] === "" && texts[1] === ""
-            // literals (including string enums) are stringified below
-            && !(types[0].flags & TypeFlags.Literal)
-            // infer T extends StringLike can't be unwrapped eagerly
-            && !types[0].symbol?.declarations?.some(d => d.parent.kind === SyntaxKind.InferType)
-            && isTypeAssignableTo(types[0], stringType)
-        ) {
+        if (texts.length === 2 && texts[0] === "" && texts[1] === "" && (types[0].flags & TypeFlags.StringMapping)) {

```

Comment:
no, it just keeps the behavior from https://github.com/microsoft/TypeScript/pull/55371 as this one seems like a safe reduction and i thought that it might be worth avoiding a new type identity in a case like this